common/src/Model/NodeCollection.cpp

```diff
@@ -95,8 +95,8 @@ namespace TrenchBroom {
             // that stops after finding the first brush
             for (const auto* node : m_nodes) {
                 const auto hasBrush = node->accept(kdl::overload(
-                    [](BrushNode*) -> bool { return true; },
-                    [](auto&& thisLambda, auto* other) -> bool {
+                    [](auto&&, const BrushNode*) -> bool { return true; },
```

Comment:
Note there were 2 bugs on this line:

- missing `auto&& thisLambda` causing the `auto* other` overload to be preferred. The static_assert I added catches this bug now.
- missing `const` on the BrushNode causing the `auto* other` overload to be preferred. Because the node we're calling `node->accept` on is const, it's meaningless to have non-const Node* overloads in the lambda here. I don't know how we can detect it though.

Reply:
I think we should just prevent using `auto` for the node parameter altogether. This can be done with a static_assert. These bugs are super hard to diagnose. What do you think?

Reply:
Yeah.. agreed.

Reply:
Ok, I'll make a PR for that in a moment.